packages/nx-monorepo/src/utils/common.ts

```diff
@@ -0,0 +1,123 @@
+/*! Copyright [Amazon.com](http://amazon.com/), Inc. or its affiliates. All Rights Reserved.
+SPDX-License-Identifier: Apache-2.0 */
+import { Obj, deepMerge as _deepMerge, isObject } from "projen/lib/util";
+
+/**
+ * Utility to deeply clone a value
+ * @param value Value to clone
+ * @returns Cloned value
+ */
+export function cloneDeep(value: any): any {
+  return JSON.parse(JSON.stringify(value));
+}
+
+/** Indicates if value is considered empty */
+export function isEmpty(value: any): boolean {
+  if (value == null) return true;
+  if (typeof value === "boolean") {
+    return false;
+  }
+  if (typeof value === "string") {
+    return value.length === 0;
+  }
+  if (Array.isArray(value)) {
+    return value.length === 0;
+  }
+  if (isObject(value)) {
+    return Object.keys(value).length === 0;
+  }
+  return false;
+}
+
+/** Replace empty value with undefined */
+export function asUndefinedIfEmpty(value: any): any | undefined {
+  if (isEmpty(value)) return undefined;
+  return value;
+}
+
+/** Options for deep merge function */
+export interface DeepMergeOptions {
+  /**
+   * Append array values
+   * @default false
+   */
+  readonly append?: boolean;
+  /**
+   * `undefined`s will cause a value to be deleted if destructive is enabled.
+   * @default false
+   */
+  readonly destructive?: boolean;
+}
+
+/**
+ * Recursively merge objects together into a new object.
+ *
+ * By default, arrays are not merged
+ * but overwritten just like scalars. Use the `append=true` flag to append array values.
+ *
+ * If an object is merged into a non-object, the non-object is lost.
+ *
+ * `undefined`s will cause a value to be deleted if destructive is enabled.
+ */
+export function deepMerge(
```

Comment:
Is the different to the deepMerge in projen/lib?

Reply:
Yes, it adds functionality to "append to array"... I will update description to make this clear